background-processor/src/lib.rs

```diff
@@ -0,0 +1,240 @@
+					manager.timer_chan_freshness_every_min();
+					current_time = Instant::now();
+				}
+			}
+		});
+		Self {
+			stop_thread: stop_thread_clone
+		}
+	}
+
+	pub fn stop(&self) {
+		self.stop_thread.store(true, Ordering::Relaxed)
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use bitcoin::blockdata::constants::genesis_block;
+	use bitcoin::blockdata::transaction::{Transaction, TxOut};
+	use bitcoin::network::constants::Network;
+	use lightning_persister::FilesystemPersister;
+	use lightning_persister::util;
+	use lightning::chain::chainmonitor;
+	use lightning::chain::keysinterface::{InMemoryChannelKeys, KeysManager};
+	use lightning::chain::transaction::OutPoint;
+	use lightning::get_event_msg;
+	use lightning::ln::channelmanager::{ChannelManager, SimpleArcChannelManager};
+	use lightning::ln::features::InitFeatures;
+	use lightning::ln::msgs::ChannelMessageHandler;
+	use lightning::util::config::UserConfig;
+	use lightning::util::events::{Event, EventsProvider, MessageSendEventsProvider, MessageSendEvent};
+	use lightning::util::ser::Writeable;
+	use lightning::util::test_utils;
+	use std::fs;
+	use std::mem;
+	use std::sync::{Arc, Mutex};
+	use std::time::Duration;
+	use super::BackgroundProcessor;
+
+	type ChainMonitor = chainmonitor::ChainMonitor<InMemoryChannelKeys, Arc<test_utils::TestChainSource>, Arc<test_utils::TestBroadcaster>, Arc<test_utils::TestFeeEstimator>, Arc<test_utils::TestLogger>, Arc<FilesystemPersister>>;
+
+	struct Node {
+		node: SimpleArcChannelManager<ChainMonitor, test_utils::TestBroadcaster, test_utils::TestFeeEstimator, test_utils::TestLogger>,
+		persister: Arc<FilesystemPersister>,
+		logger: Arc<test_utils::TestLogger>,
+	}
+
+	fn create_nodes(num_nodes: usize) -> Vec<Node> {
+		let mut nodes = Vec::new();
+		for i in 0..num_nodes {
+			let tx_broadcaster = Arc::new(test_utils::TestBroadcaster{txn_broadcasted: Mutex::new(Vec::new())});
+			let fee_estimator = Arc::new(test_utils::TestFeeEstimator { sat_per_kw: 253 });
+			let chain_source = Arc::new(test_utils::TestChainSource::new(Network::Testnet));
+			let logger = Arc::new(test_utils::TestLogger::with_id(format!("node {}", i)));
+			let persister = Arc::new(FilesystemPersister::new(format!("persister_{}", i)));
+			let seed = [i as u8; 32];
+			let network = Network::Testnet;
+			let now = Duration::from_secs(genesis_block(network).header.time as u64);
+			let keys_manager = Arc::new(KeysManager::new(&seed, network, now.as_secs(), now.subsec_nanos()));
+			let chain_monitor = Arc::new(chainmonitor::ChainMonitor::new(Some(chain_source.clone()), tx_broadcaster.clone(), logger.clone(), fee_estimator.clone(), persister.clone()));
+			let manager = Arc::new(ChannelManager::new(Network::Testnet, fee_estimator.clone(), chain_monitor.clone(), tx_broadcaster, logger.clone(), keys_manager.clone(), UserConfig::default(), i));
+			let node = Node { node: manager, persister, logger };
+			nodes.push(node);
+		}
+		nodes
+	}
+
+	#[test]
+	fn test_background_processor() {
```

Comment:
Little test comments :) ?

Reply:
Added a high-level description to each test, if that's what you mean :)